vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+
+The Avalanche P-Chain provides an index of every Subnet's validator set on the Avalanche Network including a BLS Public Key registered to validators (as of the [Banff Upgrade](https://github.com/ava-labs/avalanchego/releases/v1.9.0)). AWM utilizes the weighted validator sets stored on the P-Chain to build a Cross-Subnet Communication Protocol between any two Subnets on the Avalanche Network.
+
+Any Virtual Machine (VM) on Avalanche, can integrate Avalanche Warp Messaging to send and receive verifiable messages across Avalanche Subnets.
+
+## Background
+
+This README assumes familiarity with:
+
+- Avalanche P-Chain / [PlatformVM](../)
+- [ProposerVM](../../proposervm/README.md)
+- Basic familiarity with [BLS Multi-Signatures](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html)
+
+## BLS Multi-Signatures with Public-Key Aggregation
+
+Avalanche Warp Messaging utilizes BLS Multi-Signatures with Public-Key Aggregation in order to verify messages signed by another Subnet. To accomplish this, when a validator is added to the Avalanche P-Chain, it registers a BLS Public-Key to its NodeID along with a Proof of Possession of the corresponding private key to defend against a [rogue public-key attack](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html#mjx-eqn-eqaggsame).
+
+AWM utilizes the weighted validator sets registered BLS Public Keys to each validator to provide a weighted set of BLS Public Keys to verify a messaage coming from any Subnet with Warp Messaging.
+
+BLS Multi-Signatures with Public-Key Aggregation provides a way to aggregate BLS private keys, public keys, and signatures into a single key/signature.
+
+For more information on the cryptographic primitive underlying Warp, see [here](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html).
+
+## AWM Serialization
+
+Unsigned Message:
+```
++-----------------+----------+--------------------------+
+|      network_id :  uint32  |                 4 bytes  |
++-----------------+----------+--------------------------+
+| source_chain_id : [32]byte |                32 bytes  |
++-----------------+----------+--------------------------+
+|         payload :   []byte |       4 + size(payload)  |
++-----------------+----------+--------------------------+
+                             |  38 + size(payload) bytes|
+                             +--------------------------+
+```
+
+- `networkID` provides replay protection for BLS Signers across different Avalanche Networks
+- `sourceChainID` ensures that each blockchain on Avalanche can only sign a message with its own `blockchainID`
+- `payload` is a byte array containing the contents of the message. VMs can define their own message types to include in the `payload`
+
+Note: the `blockchainID` is the hash of the transaction that created the blockchain on the Avalanche P-Chain. It serves as the unique identifier for the blockchain across the Avalanche Network.
```

Comment:
We should move this above to:

```
`networkID` provides replay protection for BLS Signers across different Avalanche Networks
```

Reply:
Where do you want this moved to? This is already in the description of the `networkID` field